### ISSUE TITLE
求大佬解答，cross_results文件夹未找到

### ISSUE THREAD

**Issue by 785985821** [aker218/Baidu-2020-Language-and-Intelligent-Technology-Competition-Relation-Extraction-rank15#3]
大佬，我又来了，就是关系分类数据处理的时候，在in[79]里倒数第四行加载cls_logits变量中cross_results这个文件夹是没有的，请问是不是少了一些东西呀，谢谢解答。

**Comment by aker218** [comment]
> 大佬，我又来了，就是关系分类数据处理的时候，在in[79]里倒数第四行加载cls_logits变量中cross_results这个文件夹是没有的，请问是不是少了一些东西呀，谢谢解答。

你好，涉及到cross_results文件夹的内容是我们最后模型融合的部分，直接训练运行分类和实体识别的模型应该不需要运行这些block，我删除了ensemble部分的内容，重新上传了main.ipynb,可以重新试试。因为当时是边比赛边改。。可能执行顺序存在一些问题。

**Comment by 785985821** [comment]
> > 大佬，我又来了，就是关系分类数据处理的时候，在in[79]里倒数第四行加载cls_logits变量中cross_results这个文件夹是没有的，请问是不是少了一些东西呀，谢谢解答。
> 
> 你好，涉及到cross_results文件夹的内容是我们最后模型融合的部分，直接训练运行分类和实体识别的模型应该不需要运行这些block，我删除了ensemble部分的内容，重新上传了main.ipynb,可以重新试试。因为当时是边比赛边改。。可能执行顺序存在一些问题。

好的，谢谢解答，谢谢大佬~~~

**Comment by 785985821** [comment]
> > 大佬，我又来了，就是关系分类数据处理的时候，在in[79]里倒数第四行加载cls_logits变量中cross_results这个文件夹是没有的，请问是不是少了一些东西呀，谢谢解答。
> 
> 你好，涉及到cross_results文件夹的内容是我们最后模型融合的部分，直接训练运行分类和实体识别的模型应该不需要运行这些block，我删除了ensemble部分的内容，重新上传了main.ipynb,可以重新试试。因为当时是边比赛边

**Comment by 785985821** [comment]
> > > 大佬，我又来了，就是关系分类数据处理的时候，在in[79]里倒数第四行加载cls_logits变量中cross_results这个文件夹是没有的，请问是不是少了一些东西呀，谢谢解答。
> > 
> > 
> > 你好，涉及到cross_results文件夹的内容是我们最后模型融合的部分，直接训练运行分类和实体识别的模型应该不需要运行这些block，我删除了ensemble部分的内容，重新上传了main.ipynb,可以重新试试。因为当时是边比赛边

大佬，就是main.ipynb里的执行顺序感觉不是顺着执行的，因为我是复制到python文件里的，感觉顺序不太对，本人小白一枚，能讲下执行顺序吗？万分感谢，或者有对应的python文件吗，谢谢

**Comment by aker218** [comment]
> > > > 大佬，我又来了，就是关系分类数据处理的时候，在in[79]里倒数第四行加载cls_logits变量中cross_results这个文件夹是没有的，请问是不是少了一些东西呀，谢谢解答。
> > > 
> > > 
> > > 你好，涉及到cross_results文件夹的内容是我们最后模型融合的部分，直接训练运行分类和实体识别的模型应该不需要运行这些block，我删除了ensemble部分的内容，重新上传了main.ipynb,可以重新试试。因为当时是边比赛边
> 
> 大佬，就是main.ipynb里的执行顺序感觉不是顺着执行的，因为我是复制到python文件里的，感觉顺序不太对，本人小白一枚，能讲下执行顺序吗？万分感谢，或者有对应的python文件吗，谢谢

因为是比赛的代码，有很多中间的调整，我最近会改成python项目重新上传哈~

**Comment by 785985821** [comment]
好的，谢谢大佬------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;&quot;aker218&quot;<notifications@github.com&gt;
发送时间:&nbsp;2020年9月22日(星期二) 下午2:27
收件人:&nbsp;&quot;aker218/Baidu-2020-Language-and-Intelligent-Technology-Competition-Relation-Extraction-rank15&quot;<Baidu-2020-Language-and-Intelligent-Technology-Competition-Relation-Extraction-rank15@noreply.github.com&gt;;
抄送:&nbsp;&quot;785985821&quot;<785985821@qq.com&gt;;&quot;State change&quot;<state_change@noreply.github.com&gt;;
主题:&nbsp;Re: [aker218/Baidu-2020-Language-and-Intelligent-Technology-Competition-Relation-Extraction-rank15] 求大佬解答，cross_results文件夹未找到 (#3)

**Comment by 785985821** [comment]
> > > > > 大佬，我又来了，就是关系分类数据处理的时候，在in[79]里倒数第四行加载cls_logits变量中cross_results这个文件夹是没有的，请问是不是少了一些东西呀，谢谢解答。
> > > > 
> > > > 
> > > > 你好，涉及到cross_results文件夹的内容是我们最后模型融合的部分，直接训练运行分类和实体识别的模型应该不需要运行这些block，我删除了ensemble部分的内容，重新上传了main.ipynb,可以重新试试。因为当时是边比赛边
> > 
> > 
> > 大佬，就是main.ipynb里的执行顺序感觉不是顺着执行的，因为我是复制到python文件里的，感觉顺序不太对，本人小白一枚，能讲下执行顺序吗？万分感谢，或者有对应的python文件吗，谢谢
> 
> 因为是比赛的代码，有很多中间的调整，我最近会改成python项目重新上传哈~

就是我最近用bert模型处理这个任务，但是对于一种关系有多个三元组，比如同一条文本里对于“主演”这一关系，有（张一山，主演，家有儿女）、（杨紫，主演，家有儿女），而识别的时候我对于一种关系只有识别出一组主客体。然后我看了你的readme介绍，对于这种一对多，你采用的是每个关系识别出的多个关系主体和关系客体进行排列组合，生成不同的关系三元组。能说一下大致的思路吗？想借鉴一下，谢谢

**Comment by aker218** [comment]
> 杨紫，主演，家有儿

对于一个关系有多个主客体的情况，会进行排列组合选择出所有的可能，然后再根据距离保留5-6个(这个是阈值，可以自己调)，假设有N个主体，M个客体，共计M*N种组合，每个组合计算主体到关系词的距离+客体到关系词的距离的和，按照距离和从近到远排序，选择前6个。

**Comment by aker218** [comment]
> > > > > > 大佬，我又来了，就是关系分类数据处理的时候，在in[79]里倒数第四行加载cls_logits变量中cross_results这个文件夹是没有的，请问是不是少了一些东西呀，谢谢解答。
> > > > > 
> > > > > 
> > > > > 你好，涉及到cross_results文件夹的内容是我们最后模型融合的部分，直接训练运行分类和实体识别的模型应该不需要运行这些block，我删除了ensemble部分的内容，重新上传了main.ipynb,可以重新试试。因为当时是边比赛边
> > > 
> > > 
> > > 大佬，就是main.ipynb里的执行顺序感觉不是顺着执行的，因为我是复制到python文件里的，感觉顺序不太对，本人小白一枚，能讲下执行顺序吗？万分感谢，或者有对应的python文件吗，谢谢
> > 
> > 
> > 因为是比赛的代码，有很多中间的调整，我最近会改成python项目重新上传哈~
> 
> 就是我最近用bert模型处理这个任务，但是对于一种关系有多个三元组，比如同一条文本里对于“主演”这一关系，有（张一山，主演，家有儿女）、（杨紫，主演，家有儿女），而识别的时候我对于一种关系只有识别出一组主客体。然后我看了你的readme介绍，对于这种一对多，你采用的是每个关系识别出的多个关系主体和关系客体进行排列组合，生成不同的关系三元组。能说一下大致的思路吗？想借鉴一下，谢谢

对于一个关系有多个主客体的情况，会进行排列组合选择出所有的可能，然后再根据距离保留5-6个(这个是阈值，可以自己调)，假设有N个主体，M个客体，共计M*N种组合，每个组合计算主体到关系词的距离+客体到关系词的距离的和，按照距离和从近到远排序，选择前6个。

**Comment by 785985821** [comment]
> > > > > > > 大佬，我又来了，就是关系分类数据处理的时候，在in[79]里倒数第四行加载cls_logits变量中cross_results这个文件夹是没有的，请问是不是少了一些东西呀，谢谢解答。
> > > > > > 
> > > > > > 
> > > > > > 你好，涉及到cross_results文件夹的内容是我们最后模型融合的部分，直接训练运行分类和实体识别的模型应该不需要运行这些block，我删除了ensemble部分的内容，重新上传了main.ipynb,可以重新试试。因为当时是边比赛边
> > > > 
> > > > 
> > > > 大佬，就是main.ipynb里的执行顺序感觉不是顺着执行的，因为我是复制到python文件里的，感觉顺序不太对，本人小白一枚，能讲下执行顺序吗？万分感谢，或者有对应的python文件吗，谢谢
> > > 
> > > 
> > > 因为是比赛的代码，有很多中间的调整，我最近会改成python项目重新上传哈~
> > 
> > 
> > 就是我最近用bert模型处理这个任务，但是对于一种关系有多个三元组，比如同一条文本里对于“主演”这一关系，有（张一山，主演，家有儿女）、（杨紫，主演，家有儿女），而识别的时候我对于一种关系只有识别出一组主客体。然后我看了你的readme介绍，对于这种一对多，你采用的是每个关系识别出的多个关系主体和关系客体进行排列组合，生成不同的关系三元组。能说一下大致的思路吗？想借鉴一下，谢谢
> 
> 对于一个关系有多个主客体的情况，会进行排列组合选择出所有的可能，然后再根据距离保留5-6个(这个是阈值，可以自己调)，假设有N个主体，M个客体，共计M*N种组合，每个组合计算主体到关系词的距离+客体到关系词的距离的和，按照距离和从近到远排序，选择前6个。

好的 谢谢大佬 我去试一下

**Comment by 785985821** [comment]
> > > > > > > > 大佬，我又来了，就是关系分类数据处理的时候，在in[79]里倒数第四行加载cls_logits变量中cross_results这个文件夹是没有的，请问是不是少了一些东西呀，谢谢解答。
> > > > > > > 
> > > > > > > 
> > > > > > > 你好，涉及到cross_results文件夹的内容是我们最后模型融合的部分，直接训练运行分类和实体识别的模型应该不需要运行这些block，我删除了ensemble部分的内容，重新上传了main.ipynb,可以重新试试。因为当时是边比赛边
> > > > > 
> > > > > 
> > > > > 大佬，就是main.ipynb里的执行顺序感觉不是顺着执行的，因为我是复制到python文件里的，感觉顺序不太对，本人小白一枚，能讲下执行顺序吗？万分感谢，或者有对应的python文件吗，谢谢
> > > > 
> > > > 
> > > > 因为是比赛的代码，有很多中间的调整，我最近会改成python项目重新上传哈~
> > > 
> > > 
> > > 就是我最近用bert模型处理这个任务，但是对于一种关系有多个三元组，比如同一条文本里对于“主演”这一关系，有（张一山，主演，家有儿女）、（杨紫，主演，家有儿女），而识别的时候我对于一种关系只有识别出一组主客体。然后我看了你的readme介绍，对于这种一对多，你采用的是每个关系识别出的多个关系主体和关系客体进行排列组合，生成不同的关系三元组。能说一下大致的思路吗？想借鉴一下，谢谢
> > 
> > 
> > 对于一个关系有多个主客体的情况，会进行排列组合选择出所有的可能，然后再根据距离保留5-6个(这个是阈值，可以自己调)，假设有N个主体，M个客体，共计M*N种组合，每个组合计算主体到关系词的距离+客体到关系词的距离的和，按照距离和从近到远排序，选择前6个。
> 
> 好的 谢谢大佬 我去试一下

但是对于文本里关系词来说的话，关系词有可能不存在于文本里，那针对于这种情况应该如何处理，如何正确的选择关系词呢？

**Comment by aker218** [comment]
> > > > > > > > > 大佬，我又来了，就是关系分类数据处理的时候，在in[79]里倒数第四行加载cls_logits变量中cross_results这个文件夹是没有的，请问是不是少了一些东西呀，谢谢解答。
> > > > > > > > 
> > > > > > > > 
> > > > > > > > 你好，涉及到cross_results文件夹的内容是我们最后模型融合的部分，直接训练运行分类和实体识别的模型应该不需要运行这些block，我删除了ensemble部分的内容，重新上传了main.ipynb,可以重新试试。因为当时是边比赛边
> > > > > > 
> > > > > > 
> > > > > > 大佬，就是main.ipynb里的执行顺序感觉不是顺着执行的，因为我是复制到python文件里的，感觉顺序不太对，本人小白一枚，能讲下执行顺序吗？万分感谢，或者有对应的python文件吗，谢谢
> > > > > 
> > > > > 
> > > > > 因为是比赛的代码，有很多中间的调整，我最近会改成python项目重新上传哈~
> > > > 
> > > > 
> > > > 就是我最近用bert模型处理这个任务，但是对于一种关系有多个三元组，比如同一条文本里对于“主演”这一关系，有（张一山，主演，家有儿女）、（杨紫，主演，家有儿女），而识别的时候我对于一种关系只有识别出一组主客体。然后我看了你的readme介绍，对于这种一对多，你采用的是每个关系识别出的多个关系主体和关系客体进行排列组合，生成不同的关系三元组。能说一下大致的思路吗？想借鉴一下，谢谢
> > > 
> > > 
> > > 对于一个关系有多个主客体的情况，会进行排列组合选择出所有的可能，然后再根据距离保留5-6个(这个是阈值，可以自己调)，假设有N个主体，M个客体，共计M*N种组合，每个组合计算主体到关系词的距离+客体到关系词的距离的和，按照距离和从近到远排序，选择前6个。
> > 
> > 
> > 好的 谢谢大佬 我去试一下
> 
> 但是对于文本里关系词来说的话，关系词有可能不存在于文本里，那针对于这种情况应该如何处理，如何正确的选择关系词呢？

如果用的是阅读理解的主客体抽取方法，可以读取sub词和ob词的开头的token的start logits+结尾的token的end logits一起相加，作为这一对sub/ob的得分，按照这个得分进行排序，如果是CRF的话，也可以考虑使用sub词和ob词所有组成token的对应类的logits相加作为得分进行排序。

**Comment by 785985821** [comment]
> > > > > > > > > > 大佬，我又来了，就是关系分类数据处理的时候，在in[79]里倒数第四行加载cls_logits变量中cross_results这个文件夹是没有的，请问是不是少了一些东西呀，谢谢解答。
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > 你好，涉及到cross_results文件夹的内容是我们最后模型融合的部分，直接训练运行分类和实体识别的模型应该不需要运行这些block，我删除了ensemble部分的内容，重新上传了main.ipynb,可以重新试试。因为当时是边比赛边
> > > > > > > 
> > > > > > > 
> > > > > > > 大佬，就是main.ipynb里的执行顺序感觉不是顺着执行的，因为我是复制到python文件里的，感觉顺序不太对，本人小白一枚，能讲下执行顺序吗？万分感谢，或者有对应的python文件吗，谢谢
> > > > > > 
> > > > > > 
> > > > > > 因为是比赛的代码，有很多中间的调整，我最近会改成python项目重新上传哈~
> > > > > 
> > > > > 
> > > > > 就是我最近用bert模型处理这个任务，但是对于一种关系有多个三元组，比如同一条文本里对于“主演”这一关系，有（张一山，主演，家有儿女）、（杨紫，主演，家有儿女），而识别的时候我对于一种关系只有识别出一组主客体。然后我看了你的readme介绍，对于这种一对多，你采用的是每个关系识别出的多个关系主体和关系客体进行排列组合，生成不同的关系三元组。能说一下大致的思路吗？想借鉴一下，谢谢
> > > > 
> > > > 
> > > > 对于一个关系有多个主客体的情况，会进行排列组合选择出所有的可能，然后再根据距离保留5-6个(这个是阈值，可以自己调)，假设有N个主体，M个客体，共计M*N种组合，每个组合计算主体到关系词的距离+客体到关系词的距离的和，按照距离和从近到远排序，选择前6个。
> > > 
> > > 
> > > 好的 谢谢大佬 我去试一下
> > 
> > 
> > 但是对于文本里关系词来说的话，关系词有可能不存在于文本里，那针对于这种情况应该如何处理，如何正确的选择关系词呢？
> 
> 如果用的是阅读理解的主客体抽取方法，可以读取sub词和ob词的开头的token的start logits+结尾的token的end logits一起相加，作为这一对sub/ob的得分，按照这个得分进行排序，如果是CRF的话，也可以考虑使用sub词和ob词所有组成token的对应类的logits相加作为得分进行排序。

好的 谢谢大佬  我研究一下
